<a href="https://colab.research.google.com/github/Saetbyeol-Choi/R_Report/blob/main/Module_2_Report.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Jobs**
# Estimated Cost Of Living Based On Total Compensation
### Name: Saetbyeol Choi

## **1) Objective of the proposed research**

> The given data provides the total compensation for each job from 2013 to 2020 in two types of year types: fiscal year and calender year. Total compensation includes the value of benefits received in addition to the total salary.

> In the report, we try to understand the data in terms of inflation. Inflation refers to a situation in which prices generally continue to rise. When the income of workers, who are the main economic players, rise, the consumers' sentiment improves and consumption increases, which in turn raises prices. An increase in the cost of living due to an increase in the price of goods and services leads to an increase in income.

> Therefore, based on this phenomonon, the purpose of this report is to predict the correlation between cost of living and total compensation. In particular, we would like to infer the increase in the cost of living of employee through increases in health and dental insurance premium and increases in city contributions to retirement plans of employee.

## **2) Data Exploration**

> Load the data set and put name on the data frame as "EC".

In [ ]:
EC <- read.csv("/content/employee-compensation.csv")

##### 1. Showing the first 10 rows

> Use square brackets "[]" to subset 10 rows from the data frame.
By typing 1:10, the rows from 1 to 10 are extracted.

In [ ]:
EC[1:10,]

Organization.Group.Code Job.Family.Code Job.Code Year.Type Year
1  1                       1000            1021     Calendar  2013
2  1                       1000            1023     Calendar  2013
3  1                       1000            1031     Calendar  2013
4  1                       1000            1054     Calendar  2013
5  1                       1000            1062     Calendar  2013
6  1                       1000            1064     Calendar  2013
7  1                       1000            1070     Calendar  2013
8  1                       1000            1094     Calendar  2013
9  1                       1200            1232     Calendar  2013
10 1                       1600            1654     Calendar  2013
   Organization.Group Department.Code Department          Union.Code
1  Public Protection  ADP             ADP Adult Probation 21        
2  Public Protection  ADP             ADP Adult Probation 21        
3  Public Protection  ADP             ADP Adult Probation 21        
4  Public Protection  ADP             ADP Adult Probation 21        
5  Public Protection  ADP             ADP Adult Probation 21        
6  Public Protection  ADP             ADP Adult Probation 21        
7  Public Protection  ADP             ADP Adult Probation 21        
8  Public Protection  ADP             ADP Adult Probation 21        
9  Public Protection  ADP             ADP Adult Probation 21        
10 Public Protection  ADP             ADP Adult Probation 21        
   Union                                           ⋯ Employee.Identifier
1  Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 37730              
2  Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 26574              
3  Prof & Tech Engineers - Miscellaneous, Local 21 ⋯  8148              
4  Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 27436              
5  Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 37730              
6  Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 36223              
7  Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 51426              
8  Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 26574              
9  Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 37983              
10 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 16858              
   Salaries  Overtime Other.Salaries Total.Salary Retirement Health.and.Dental
1   57534.65 0            0.00        57534.65    11253.16   11961.34         
2   57678.50 0            0.00        57678.50    10284.88    7176.80         
3   63532.93 0            0.00        63532.93    12495.83   12918.24         
4  101274.51 0        -7058.59        94215.92    19644.45   12918.24         
5    5084.00 0            0.00         5084.00     1083.90     956.91         
6   34456.00 0            0.00        34456.00     7346.00    3827.63         
7  139889.00 0            0.00       139889.00    26799.23   12918.24         
8   46572.00 0            0.00        46572.00     9929.16    5741.44         
9   84681.53 0            0.00        84681.53    16661.84   12918.24         
10  10611.01 0            0.00        10611.01     1879.20    1435.36         
   Other.Benefits Total.Benefits Total.Compensation
1  4532.63        27747.13        85281.78         
2  4755.14        22216.82        79895.32         
3  4702.71        30116.78        93649.71         
4  7458.76        40021.45       134237.37         
5   387.50         2428.31         7512.31         
6  2579.95        13753.58        48209.58         
7  9791.54        49509.01       189398.01         
8  3822.28        19492.88        66064.88         
9  6838.44        36418.52       121100.05         
10  827.09         4141.65        14752.66

##### 2. Showing the number of rows and columns

> By using "dim()' function, the number of rows and columns are found to be 683,277 and 22, respectively.


In [ ]:
dim(EC)

[1] 683277     22

##### 3. Showing types of columns (using str)
> The "str()" function simply summarizes the contents of the data frame. It helps to understand the data frame by outputting the data type of each column, including the number of rows and columns that can be checked with the "dim()" function.<br> The data type(chr, num, int) and data value of each column(total 22) are checked with the "str()" function.

In [ ]:
str(EC)

'data.frame':	683277 obs. of  22 variables:
 $ Organization.Group.Code: int  1 1 1 1 1 1 1 1 1 1 ...
 $ Job.Family.Code        : chr  "1000" "1000" "1000" "1000" ...
 $ Job.Code               : chr  "1021" "1023" "1031" "1054" ...
 $ Year.Type              : chr  "Calendar" "Calendar" "Calendar" "Calendar" ...
 $ Year                   : int  2013 2013 2013 2013 2013 2013 2013 2013 2013 2013 ...
 $ Organization.Group     : chr  "Public Protection" "Public Protection" "Public Protection" "Public Protection" ...
 $ Department.Code        : chr  "ADP" "ADP" "ADP" "ADP" ...
 $ Department             : chr  "ADP Adult Probation" "ADP Adult Probation" "ADP Adult Probation" "ADP Adult Probation" ...
 $ Union.Code             : int  21 21 21 21 21 21 21 21 21 21 ...
 $ Union                  : chr  "Prof & Tech Engineers - Miscellaneous, Local 21" "Prof & Tech Engineers - Miscellaneous, Local 21" "Prof & Tech Engineers - Miscellaneous, Local 21" "Prof & Tech Engineers - Miscellaneous, Local 21

##### 4. Showing median of specific columns

> The value of average can be checked by using the "median()" function, and the average of total compensation is 107,529.24. 



In [ ]:
median(EC$Total.Compensation, na.rm=TRUE)

[1] 107529.2

## **3) Subsets of Data**


> Run library(dplyr) for data manipulation. 

In [ ]:
library(dplyr)

> Analyze the fiscal year as the base year in order to increase the consistency of the data. <br>

> The "which()" function indicates information about the position of data that satisfies the condition. The "which()" function is used to select the values corresponding to the fiscal year among the year types. Newly selected data frame is named "FEC" and listed in order of greatest total compensation. Use the "head()" function to extract the largest total compensation value from 1 to 5, containing distinct rows and columns. In this case, the columns are selected from Year to Total Compensation.

> From the output, it can be seen that the total compensation for a worker with a job title of Police Officer 3 in 2019 is extracted with the highest value of 807,625.2. Based on this, we would like to extract the largest value of total compensation of previous years and compare them with the output value of 2019.

In [ ]:
FEC <- arrange(EC[which(EC$Year.Type == "Fiscal"),],desc(Total.Compensation))
FEC_h <- head(FEC[,5:22],5)
FEC_h

Year Organization.Group               Department.Code Department         
1 2019 Public Protection                POL             Police             
2 2020 General Administration & Finance RET             Retirement Services
3 2019 General Administration & Finance RET             Retirement Services
4 2018 General Administration & Finance RET             Retirement Services
5 2019 Public Protection                FIR             Fire Department    
  Union.Code Union                      Job.Family                   
1 911        POA                        Police Services              
2 351        Municipal Exec Assoc, Misc Administrative & Mgmt (Unrep)
3 351        Municipal Exec Assoc, Misc Administrative & Mgmt (Unrep)
4 351        Municipal Exec Assoc, Misc Administrative & Mgmt (Unrep)
5 352        Municipal Exec Assoc, Fire Management                   
  Job                         Employee.Identifier Salaries Overtime
1 Police Officer 3            37499756            651936.7 3747    
2 Chief Investment Officer    42466186            584297.6    0    
3 Chief Investment Officer    37489389            565707.0    0    
4 Chief Investment Officer    36698179            550065.9    0    
5 Dep Chf of Dept (Fire Dept) 40266669            262160.9    0    
  Other.Salaries Total.Salary Retirement Health.and.Dental Other.Benefits
1   3183.86      658867.6     121952.52  15525.74          11279.42      
2      0.00      584297.6     114934.16  18155.15          18555.87      
3      0.00      565707.0     105697.07  17788.88          17937.01      
4      0.00      550065.9     100845.76  17294.69          17395.43      
5 560099.40      591910.5      50184.86  17368.33           9432.78      
  Total.Benefits Total.Compensation
1 148757.68      807625.2          
2 151645.18      735942.8          
3 141422.96      707130.0          
4 135535.88      685601.7          
5  76985.97      668896.5

> The largest value of total compensation in 2018 is extracted using the "which()" function, and the columns are selected from Year to Total Compensation.The job with the largest total compensation in 2018 is Chief Investment Officer, with an amount of 685,601.7. This shows that the amount of total compensation in 2018 is smaller than the one in 2019.

In [ ]:
head(FEC[which(FEC$Year == 2018),5:22],5)

Year Organization.Group               Department.Code Department         
4  2018 General Administration & Finance RET             Retirement Services
15 2018 General Administration & Finance RET             Retirement Services
18 2018 General Administration & Finance RET             Retirement Services
24 2018 Community Health                 DPH             Public Health      
28 2018 Public Protection                FIR             Fire Department    
   Union.Code Union                          Job.Family                   
4  351        Municipal Exec Assoc, Misc     Administrative & Mgmt (Unrep)
15 351        Municipal Exec Assoc, Misc     Unassigned                   
18 351        Municipal Exec Assoc, Misc     Unassigned                   
24 351        Municipal Exec Assoc, Misc     Administrative & Mgmt (Unrep)
28 799        Firefighters,Local 798, Unit 2 Fire Services                
   Job                            Employee.Identifier Salaries Overtime
4  Chief Investment Officer       36698179            550065.9      0  
15 Managing Director              36818957            460025.9      0  
18 Managing Director              37305193            460025.9      0  
24 Physician Administrator, DPH   36875497            426657.9      0  
28 Battalion Chief, Fire Suppress 36776890            207432.6 111595  
   Other.Salaries Total.Salary Retirement Health.and.Dental Other.Benefits
4       0.0       550065.9     100845.76  17294.69          17395.43      
15      0.0       460025.9      84338.41  23081.84          16127.64      
18      0.0       460025.9      84338.41   8456.90          15902.16      
24      0.0       426657.9      79909.54  18210.36          15484.42      
28 208126.1       467547.1      35524.68  19232.67           7435.76      
   Total.Benefits Total.Compensation
4  135535.88      685601.7          
15 123547.89      583573.8          
18 108697.47      568723.3          
24 113604.32      540262.2          
28  62193.11      529740.2

> In the same way as above, the rows and columns with the largest value of total compensation in 2017 is extracted. The highet value in total compensation is 668,412.4, which is smaller than in 2019 and 2018. The amount is from the same job category as in 2018.

> It can be seen that total compensation increased from 2017 to 2019. In the case of FY2017 and FY2018, it can be inferred that income increase may have occurred due to an increase in the cost of living, considering that there was an increase in all aspects of salaries and benefits in the same job group.

In [ ]:
head(FEC[which(FEC$Year == 2017),5:22],5)

Year Organization.Group               Department.Code Department           
6  2017 General Administration & Finance RET             RET Retirement System
7  2017 General Administration & Finance RET             Retirement Services  
25 2017 Community Health                 DPH             DPH Public Health    
26 2017 Community Health                 DPH             Public Health        
27 2017 General Administration & Finance RET             RET Retirement System
   Union.Code Union                                          
6  351        Municipal Executive Association - Miscellaneous
7  351        Municipal Exec Assoc, Misc                     
25 351        Municipal Exec Assoc-Misc                      
26 351        Municipal Exec Assoc, Misc                     
27 351        Municipal Exec Assoc-Misc                      
   Job.Family                    Job                         
6  Administrative & Mgmt (Unrep) Chief Investment Officer    
7  Administrative & Mgmt (Unrep) Chief Investment Officer    
25 Administrative & Mgmt (Unrep) Physician Administrator, DPH
26 Administrative & Mgmt (Unrep) Physician Administrator, DPH
27 Administrative & Mgmt (Unrep) Managing Director           
   Employee.Identifier Salaries Overtime Other.Salaries Total.Salary Retirement
6     25997            533985.9 0        0              533985.9     95600.00  
7  37074780            532413.1 0        0              532413.1     95361.27  
25     1110            418016.6 0        0              418016.6     74197.99  
26 37210666            425455.8 0        0              425455.8     75559.62  
27    45946            420076.6 0        0              420076.6     75146.60  
   Health.and.Dental Other.Benefits Total.Benefits Total.Compensation
6  13765.55          25060.92       134426.5       668412.4          
7  16606.61          17016.45       128984.3       661397.5          
25 13185.38          33391.73       120775.1       538791.7          
26 17396.53          19610.29       112566.4       538022.2          
27 13765.55          23701.87       112614.0       532690.6

> I would like to extract the employees who received the highest benefits in terms of city-paid premiums to health and dental insurance plans and the contributions to employee retirement plans, which are essential benefits for living.

> Sort the data frame in descending order based on the retirement column using the "arrange()" function with "desc()". By using "head()" function, the lartest value of the retirement is extracted. 

> It is shown that the job with the highest value of retirement is Police Officer 3, which received 121,952.5 and it is in 2019, the same year as the year with the largest total compensation.

In [ ]:
Ret <- arrange(FEC[,5:22],desc(Retirement))
Ret_h <- head(Ret,5)
Ret_h

Year Organization.Group               Department.Code Department           
1 2019 Public Protection                POL             Police               
2 2020 General Administration & Finance RET             Retirement Services  
3 2019 General Administration & Finance RET             Retirement Services  
4 2016 General Administration & Finance RET             RET Retirement System
5 2018 General Administration & Finance RET             Retirement Services  
  Union.Code Union                                          
1 911        POA                                            
2 351        Municipal Exec Assoc, Misc                     
3 351        Municipal Exec Assoc, Misc                     
4 351        Municipal Executive Association - Miscellaneous
5 351        Municipal Exec Assoc, Misc                     
  Job.Family                    Job                      Employee.Identifier
1 Police Services               Police Officer 3         37499756           
2 Administrative & Mgmt (Unrep) Chief Investment Officer 42466186           
3 Administrative & Mgmt (Unrep) Chief Investment Officer 37489389           
4 Administrative & Mgmt (Unrep) Chief Investment Officer    25997           
5 Administrative & Mgmt (Unrep) Chief Investment Officer 36698179           
  Salaries Overtime Other.Salaries Total.Salary Retirement Health.and.Dental
1 651936.7 3747     3183.86        658867.6     121952.5   15525.74         
2 584297.6    0        0.00        584297.6     114934.2   18155.15         
3 565707.0    0        0.00        565707.0     105697.1   17788.88         
4 515101.8    0        0.00        515101.8     101306.0   13067.98         
5 550065.9    0        0.00        550065.9     100845.8   17294.69         
  Other.Benefits Total.Benefits Total.Compensation
1 11279.42       148757.7       807625.2          
2 18555.87       151645.2       735942.8          
3 17937.01       141423.0       707130.0          
4 24021.54       138395.5       653497.3          
5 17395.43       135535.9       685601.7

> After sorting the data frame called "FEC" in descending order by the column of health and dental in the same way as above, the value with the highest number on the column is extracted. The value is 52,146.51 in 2019, which comes from a Manager VI.

> Compare this with the Police Officer 3 who has the largest value of total compensation and retirement in 2019. The manager received a lot of benefits on helath and dental, but their basic salary is lower than that of the police and there was no overtime. As a result, it appears that the manager's total compensation is lower than that of the police. 

> Based on this, it can be seen that the benefits that are important to each job are different, and factors affecting the total compensation are vary.

In [ ]:
HD <- arrange(FEC[5:22],desc(Health.and.Dental))
HD_h <- head(HD,5)
HD_h

Year Organization.Group                       Department.Code
1 2019 Public Works, Transportation & Commerce  PUC            
2 2020 Public Works, Transportation & Commerce  PUC            
3 2018 Public Protection                        CRT            
4 2020 Human Welfare & Neighborhood Development CHF            
5 2019 Public Protection                        CRT            
  Department                  Union.Code Union                      Job.Family
1 Public Utilities Commission 351        Municipal Exec Assoc, Misc Management
2 Public Utilities Commission 351        Municipal Exec Assoc, Misc Management
3 Trial Courts                356        Court-MEA                  Untitled  
4 Children Youth & Families   351        Municipal Exec Assoc, Misc Management
5 Trial Courts                356        Court-MEA                  Untitled  
  Job                 Employee.Identifier Salaries  Overtime Other.Salaries
1 Manager VI          39115976            176673.70 0         0            
2 Manager VI          44086226            200745.02 0         0            
3 Court Supervisor II 37140345             70466.32 0         0            
4 Manager III         42476363            159102.11 0         0            
5 Court Supervisor II 38776095             99828.94 0        72            
  Total.Salary Retirement Health.and.Dental Other.Benefits Total.Benefits
1 176673.70    32906.51   52146.51          15573.06       100626.08     
2 200745.02    40917.79   49000.22          12676.46       102594.47     
3  70466.32    13564.82   44921.80           6596.75        65083.37     
4 159102.11    32428.06   43446.48          13713.49        89588.03     
5  99900.94    19096.27   42616.85           9407.39        71120.51     
  Total.Compensation
1 277299.8          
2 303339.5          
3 135549.7          
4 248690.1          
5 171021.5

## **4) Single Vector**



> List the jobs based on health insurance premiums related to medical insurance, which is one of the important benefits in life.


> Use the "which()" function to extract the data corresponding to the year of 2020. Use the "filter()" function to exclude missing values from the column of health and dental and sort them with the "arrange()" function. Beacuse the defaults of the "arrange()" function is in ascending order, use "desc()" to sort in descending order based on the amount of health and dental. By using the "head()" function, the 5 rows which have the largest values in the column of health and dental are selected and the data frame is given a new name "JOB".

In [ ]:
JOB <- head(arrange(FEC[which(FEC$Year == 2020),5:22] %>% filter(is.na(Health.and.Dental) == FALSE), desc(Health.and.Dental)),5)
JOB

Year Organization.Group                       Department.Code
1 2020 Public Works, Transportation & Commerce  PUC            
2 2020 Human Welfare & Neighborhood Development CHF            
3 2020 Public Protection                        CRT            
4 2020 Public Protection                        CRT            
5 2020 Public Protection                        DEM            
  Department                   Union.Code Union                      Job.Family
1 Public Utilities Commission  351        Municipal Exec Assoc, Misc Management
2 Children Youth & Families    351        Municipal Exec Assoc, Misc Management
3 Trial Courts                 356        Court-MEA                  Untitled  
4 Trial Courts                 356        Court-MEA                  Untitled  
5 Dept of Emergency Management 351        Municipal Exec Assoc, Misc Management
  Job                        Employee.Identifier Salaries Overtime
1 Manager VI                 44086226            200745.0 0       
2 Manager III                42476363            159102.1 0       
3 Court Computer Sys Manager 43416109            139009.3 0       
4 Court Manager              43966465            115765.4 0       
5 Manager I                  42476256            137451.1 0       
  Other.Salaries Total.Salary Retirement Health.and.Dental Other.Benefits
1    0.00        200745.0     40917.79   49000.22          12676.46      
2    0.00        159102.1     32428.06   43446.48          13713.49      
3 9133.04        144793.3     29015.67   35633.00          12701.09      
4 7084.72        120525.2     24154.33   35633.00          10662.12      
5    0.00        137451.1     28040.23   33480.25          10662.26      
  Total.Benefits Total.Compensation
1 102594.47      303339.5          
2  89588.03      248690.1          
3  77349.76      222143.1          
4  70449.45      190974.7          
5  72182.74      209633.8

> By using "as.vector()", convert the values of the selected column into a vector and it's named "v".<br>
The jobs with the highest value on the benefits of health and dental insurance premium in 2020 are as follows, in order.
> For those who do not have health insurance, personal medical expenses will increase. When medical expenses go up, insurance premiums also go up. This is directly related to the cost of living, which increases the individual's cost of living. By comparing the medical expenses of the relevant jobs extracted with the previous year, the increase or decrease in the cost of living can be inferred.

In [ ]:
v <- as.vector(JOB$Job)
v

[1] "Manager VI"                 "Manager III"               
[3] "Court Computer Sys Manager" "Court Manager"             
[5] "Manager I"

> It is checked that the "v" is a vector by calling "is.vector()" fuunction.

In [ ]:
 is.vector(v)

[1] TRUE

> A data frame to check the amount of health and dental insurance premium from 2013 to 2019 for the same jobs extracted from 2020 can be created and compared with 2020.

> By using the %in% operator, which determines if an element belongs to a data frame, with the "which()" function, data from 2013 to 2019 for the 5 jobs selected above are extracted. After sorting by the column of health and dental, the 5 rows corresponding to the highest values are extracted.

> Except for the fact that Manager VI's health insurance premium in 2019 is the highest at 52,146.51, it can be seen that health insurance premium in other years are lower than in 2020. By comparing them, it can be estimated that the cost of living related to health insurance premiums increased in 2020. It also can be inferred that the relationship between total compensation and health insurance premium has somewhat positive correlation.

In [ ]:
head(arrange(FEC[which(FEC$Year %in% 2013:2019 & FEC$Job %in% c('Manager VI','Manager III','Court Computer Sys Manager','Court Manager','Manager I')),5:22],
desc(Health.and.Dental)),5)

Year Organization.Group                      Department.Code
1 2019 Public Works, Transportation & Commerce PUC            
2 2019 General Administration & Finance        RET            
3 2019 Public Protection                       CRT            
4 2019 Public Protection                       CRT            
5 2019 Public Protection                       CRT            
  Department                  Union.Code Union                      Job.Family
1 Public Utilities Commission 351        Municipal Exec Assoc, Misc Management
2 Retirement Services         351        Municipal Exec Assoc, Misc Management
3 Trial Courts                356        Court-MEA                  Untitled  
4 Trial Courts                356        Court-MEA                  Untitled  
5 Trial Courts                356        Court-MEA                  Untitled  
  Job                        Employee.Identifier Salaries Overtime
1 Manager VI                 39115976            176673.7 0       
2 Manager VI                 40616130            156112.7 0       
3 Court Manager              38775982            127465.0 0       
4 Court Computer Sys Manager 39756315            155740.5 0       
5 Court Manager              36655716            127465.0 0       
  Other.Salaries Total.Salary Retirement Health.and.Dental Other.Benefits
1    0           176673.7     32906.51   52146.51          15573.06      
2    0           156112.7     29076.07   29980.84          11185.01      
3    0           127465.0     23741.15   29738.23          10861.81      
4 5877           158679.0     29007.76   29639.29          11001.40      
5 4810           129870.0     23741.15   29639.29          10904.81      
  Total.Benefits Total.Compensation
1 100626.08      277299.8          
2  70241.92      226354.6          
3  64341.19      191806.2          
4  69648.45      228327.5          
5  64285.25      194155.2

## **5) 5 different Lists**

> Based on the 5 job titles extracted in the order of highest health insurance premium in 2020, this time, the amount in the retirement column for each job is sorted in descending order.

> Use the "which()" fuction to extract the values corresponding to retirement column among the data corresponidng to each job titles. It is sorted in descending order by using the "sort()" function with the "decreasing=TRUE" argument. The values corresponding to the top 5 can be extracted from each data frame by using the "head()" function. Through the "list()" function, each vector from job1 to job5 is classified by job titles and the result is returned as a list.

> Manager VI who had the largest amount on the health insurance premium benefit also seemed to have the highest amount on the retirement benefit among all jobs extracted above. From the results, it can be seen that jobs with high health insurance premium also have high value on the retirement column. They may not always proportional based on the other jobs cases but it can be seen that they are related. In addition, it can be seen that the benefits that are important to each job are different. 

In [ ]:
job1 <- head(sort(FEC[which(FEC$Job == "Manager VI"),"Retirement"], decreasing = TRUE),5)
job2 <- head(sort(FEC[which(FEC$Job == "Manager III"),"Retirement"], decreasing = TRUE),5)
job3 <- head(sort(FEC[which(FEC$Job == "Court Computer Sys Manager"),"Retirement"], decreasing = TRUE),5)
job4 <- head(sort(FEC[which(FEC$Job == "Court Manager"),"Retirement"], decreasing = TRUE),5)
job5 <- head(sort(FEC[which(FEC$Job == "Manager I"),"Retirement"], decreasing = TRUE),5)
list("Manager VI"=job1, "Manager III"=job2, "Court Computer Sys Manager"=job3, "Court Manager"=job4, "Manager I"=job5)

$`Manager VI`
[1] 49556.45 48198.12 47477.12 46398.54 46260.89

$`Manager III`
[1] 38082.20 38043.33 37712.34 37635.31 37549.99

$`Court Computer Sys Manager`
[1] 32680.33 32680.33 31472.24 31012.02 31012.02

$`Court Manager`
[1] 26753.54 26753.54 26753.54 26753.54 26753.54

$`Manager I`
[1] 35588.94 34139.32 33712.37 33401.11 33356.09

## **6) Conditionals**

> The "if()", "else if()", and "else()" functions used when there are multiple conditions execute and return the contents when the conditions are met. By applying the conditional expressions, the increase or decrease of total compensation by year can be checked. 
The condition is divided into three cases, such as when two different years are entered and the value is less than, greater than, or equal to. If condition 1 is satisfied, "increased" is returned, and if condition 2 is satisfied, "decreased" is returned. If both of these conditions are not satisfied, "non related" is returned.

In [ ]:
degree <- function(year1, year2) {
  result <- if ((head(FEC[which(FEC$Year == year1),22],1)) < head(FEC[which(FEC$Year == year2),22],1)) "increased"
  else if ((head(FEC[which(FEC$Year == year1),22],1)) > head(FEC[which(FEC$Year == year2),22],1)) "decreased"
  else if ((head(FEC[which(FEC$Year == year1),22],1)) == head(FEC[which(FEC$Year == year2),22],1)) "did not change"
  else "non related"
  cat("The total compensation", result, "in", year2, "compared to" ,year1, "\n")
  }

> By putting two years, the corresponding conditional statement is returned. It is confirmed that total compensation has continued to increase except in 2020.

In [ ]:
degree(2013, 2014)
degree(2014, 2015)
degree(2015, 2016)
degree(2016, 2017)
degree(2017, 2018)
degree(2018, 2019)
degree(2019, 2020)

The total compensation increased in 2014 compared to 2013 
The total compensation increased in 2015 compared to 2014 
The total compensation increased in 2016 compared to 2015 
The total compensation increased in 2017 compared to 2016 
The total compensation increased in 2018 compared to 2017 
The total compensation increased in 2019 compared to 2018 
The total compensation decreased in 2020 compared to 2019 


> Use "which()" function one of the conditional expressions to extract the year with the "sum()" function that calculates the sum to find the increment of total compensation between 2013 and 2020. The total amount of total compensation increased by 1,678,062,899.68, which is 1.44% higher than in 2013.

In [ ]:
diffT <- sum(FEC[which(FEC$Year == 2020),22]) - sum(FEC[which(FEC$Year == 2013),22])
sumT <- sum(FEC[which(FEC$Year == 2020),22]) / sum(FEC[which(FEC$Year == 2013),22])
diffT
sumT

[1] 1678062900

[1] 1.442391

> Compare the overall average with the average from each year from 2013 to 2020 to find out the increase or decrease. Find the overall average of total compensation by using "mean()" function and named it "TCav". Set "max, min, year" to assign the vlaue on them. Assign the value obtained by dividing the total compensation average of the year by the total average to "min". Use "repeat()" function to make it loopable from 2013 to 2020. By using "break", the execution will be stopped when it's applied to 2021. Use the "if()" function to assign a condition. By using "switch()" function that returns a specific output to a specific input, the 1 assigned to z, the first execution statement, comes out.<br> The year with the highest increase or decrease rate is 2020, which is 1.205%.

In [ ]:
TCav <- mean(FEC$Total.Compensation, na.rm=TRUE)
max <- 0 # to assign value 
min <- 0 
year <- 0
i<-2013
repeat{
  min = mean(FEC[which(FEC$Year == i),22])/TCav
  z <- 1
  if(max > min)
  {
    z <- 2
  }
 switch(z,max <- min,max <- max)
 switch(z,year <- i,year<-year)
 i<-i+1
 if( i < 2021) next
 else break
}
print(max)
print(year)

[1] 1.205204
[1] 2020


## 7) Loops

> By using the "for()" function that repeats a specific part a set number of times, it can be checked whether the amount on the column of retirement increases or decreases by year. The "as.integer()" function converts a character object to an integer object so that it can be determined that whether it is increasing or decreasing.


> Comparing each year from 2013 to 2022, it is shown that the amount on the retirement column decreased compared to the previous year only in the case of 2020 and 2017. 


In [ ]:
for(i in 2013:2019)
{
  diffR = as.integer((head(Ret[which(Ret$Year == i+1),14],1))) - as.integer((head(Ret[which(Ret$Year == i),14],1)))
  if(diffR > 0) {
    cat("The contributions to employee retirement in", i+1, "increased compared to", i,"\n")
  }else 
  cat("The contributions to employee retirement in", i+1, "decreased compared to", i,"\n")
  }

The contributions to employee retirement in 2014 increased compared to 2013 
The contributions to employee retirement in 2015 increased compared to 2014 
The contributions to employee retirement in 2016 increased compared to 2015 
The contributions to employee retirement in 2017 decreased compared to 2016 
The contributions to employee retirement in 2018 increased compared to 2017 
The contributions to employee retirement in 2019 increased compared to 2018 
The contributions to employee retirement in 2020 decreased compared to 2019 


> Use the "while()" function suitable for infinite repetition processing to check the increase or decrease of the amount on the column of health and dental. Set the value of "i" so that the iteration starts in 2013 and stops in 2020. The amount continued to increase except in 2015 and 2020.

In [ ]:
i = 2013
while(i < 2020){
  {diffH = as.integer((head(HD[which(HD$Year == i+1),15],1))) - as.integer((head(HD[which(HD$Year == i),15],1)))
  cat("The differnce between", i+1, "and", i, "is", diffH, "\n") 
  i=i+1
  }}

The differnce between 2014 and 2013 is 4627 
The differnce between 2015 and 2014 is -310 
The differnce between 2016 and 2015 is 1711 
The differnce between 2017 and 2016 is 13874 
The differnce between 2018 and 2017 is 9176 
The differnce between 2019 and 2018 is 7225 
The differnce between 2020 and 2019 is -3146 



> The ratio of the year average to the total average of total compensation can be calculated and compared. The "repeat()" function is used to get the ratio from 2013 to 2020. The year in which the ratio of that year's average to the overall total compensation was high is 2020, and the value is 1.205%. This is the same sas the value checked above.

In [ ]:
TCav <- mean(FEC$Total.Compensation, na.rm=TRUE)
i <- 2012
repeat{
  rate = mean(FEC[which(FEC$Year == i+1),22])/TCav
  i <- i+1
  if( i == 2021) break
  cat("The ratio on", i, "is", rate, "\n")}

The ratio on 2013 is 0.8969418 
The ratio on 2014 is 0.9003178 
The ratio on 2015 is 0.9055658 
The ratio on 2016 is 0.9302701 
The ratio on 2017 is 0.992359 
The ratio on 2018 is 1.078879 
The ratio on 2019 is 1.087195 
The ratio on 2020 is 1.205204 


## **8) Summary**

> A fiscal year is a 12-month accounting period that a business uses for financial and tax reporting purposes. The report rearranged the data based on these fiscal years to increase the unity of data and the accuracy of comparative analysis.<br>
> Based on the total compensation that includes both annual salary and benefits, the data were compared by year and job titles. After 2013 and before 2020, it was confirmed that total compensation has been increased overall. The largest total compensation was in 2019 and it is beacuse Police Officer 3 held the highest value on total compensation. By extracting and comparing the amount on the column of retirement and health and dental, it was found that they also have been increased overall by year. The job with the highest total compensation was the same as the job with the highest value on the column of retiremnt, but it was different from the job with the highest value on the column of health and dental. Based on this it can be seen that the proportion of each benefit in the total compensation vaires. Also, through the oveerall increase in retirement, it can be inferred that people's essential retirement funds after retirement continue to increase compared to the previous year.

## **9) Recommendations**

> In times of inflation, workers expect income increases, and companies accelerate the price rise by rising product prices to reflect labor costs and prices. In order for pices to rise consistently and generally, income incluing wages or salary must rise. In this way, income increase is both a cause and a result of inflation. Unless income increases, the standard of living will inevitably fall. Therefore, it can be seen that the cost of living and income are related to each other.<br>
Based on these facts, it can be said that total compensation continued to rise from 2013 to 2019 from the data, and thus the cost of living also increased during the period. In addition, it can be inferred that health insurance premiums directly related to health and people's essential retirement funds after retirement continues to increase overall.<br>
As a result, it can be said that it is necessary to strike a balance between prices and income so that the negative effects of inflation do not widen the gap between the cost of living and income or overestimated. Another fact is that total compensation is correlated with, but not always proporional to, the benefits of retirement and health and dental. This may imply that factors considered by jobs may differ between salaries and benefits. 